src/worker.rs

```diff
@@ -149,6 +166,20 @@ impl Worker {
                         } else {
                             None
                         };
+                        let resp = if mutated && disallowed_mutation(allowed_to_mutate) {
```

Comment:
@ereslibre , your suggestion to make this `if` more readable does not compile due a issue of comparing different types (`&mut Option<bool>` and `Option<bool>`). One solution is to implement the `PartialEq` trait for the types. But I was not sure if implementing the trait for type from std lib is a best practice or not. What do you think? 

Anyway, trying to improve the readability I've extracted the code into a function.

Reply:
You can do:

```rust
let resp = if mutated && allowed_to_mutate.as_ref() == Some(&false) {
```

I think is a bit more clear than the utility function.

Reply:
:facepalm: I need to put more rust in daily routine...

Reply:
Done!